src/Eloquent/NeoEloquentFactory.php

```diff
@@ -0,0 +1,257 @@
+<?php namespace Vinelab\NeoEloquent\Eloquent;
```

Comment:
we shouldn't be re-writing factories. Could there be a way to reuse the one that's shipped with Laravel by introducing compatibility to it? And avoid providing a separate method `neo_factory` to rely on the built-in one.

Reply:
@Mulkave I was looking for a way to avoid re-writing factories but I didn't find a way to do this easier than this solution.


**SOLUTION description:**
We had next issues:
1. Wrong check for Model instance that is used in default `Illuminate\Database\Eloquent\FactoryBuilder.php` thus when we try to create 1 model, it thinks that we try to create a collection of models and we get the `each() method does not exists...`.
```php
    /**
     * Create a collection of models and persist them to the database.
     *
     * @param  array  $attributes
     * @return mixed
     */
    public function create(array $attributes = [])
    {
        $results = $this->make($attributes);

        if ($results instanceof Model) {
            $this->store(collect([$results]));
        } else {
            $this->store($results);
        }

        return $results;
    }
```

2. Bug in default `Illuminate\Database\Eloquent\FactoryBuilder.php` (store method) about `non-existing getName() method...`
Was:
```php
protected function store($results)
    {
        $results->each(function ($model) {
            if (! isset($this->connection)) {

                $model->setConnection($model->newQueryWithoutScopes()->getConnection()->getName());
            }

            $model->save();
        });
    }

should be:
**FactoryBuilder→**
```php
/**
 * Set the connection name on the results and store them.
 *
 * @param  \Illuminate\Support\Collection  $results
 * @return void
 */
protected function store($results)
{
    $results->each(function ($model) {
        if (! isset($this->connection)) {
            $model->setConnection($model->getConnectionName());
        }
        $model->save();
    });
}
```

Reply:
@Mulkave take a look at a new update. New classes extend default Laravel classes and we only re-implement 2 methods that cause issues.

Reply:
nice! Getting there. I see that it's no longer extending `FactoryBuilder` but instead implementation `ArrayAccess` interface. I'm thinking of the limitations this introduces when there are updates on Laravel's side. It would be easier to adopt updates if we were to still extend `FactoryBuilder`.

Reply:
maybe you looked at outdated version. 

`class NeoFactoryBuilder extends FactoryBuilder`

Reply:
ah, actually i was looking at the wrong class `NeoEloquentFactory`. Yes, what about that class?

Reply:
`NeoEloquentFactory` can't extend `Illuminate\Database\Eloquent\Factory` because we are replacing default `Factory`:

```php
        $this->app->booting(function () {
            $loader = \Illuminate\Foundation\AliasLoader::getInstance();
            $loader->alias('NeoEloquent', 'Vinelab\NeoEloquent\Eloquent\Model');
            $loader->alias('Neo4jSchema', 'Vinelab\NeoEloquent\Facade\Neo4jSchema');
            **$loader->alias('Illuminate\Database\Eloquent\Factory', 'Vinelab\NeoEloquent\Eloquent\NeoEloquentFactory');**
        });